src/gaxios.ts

```diff
@@ -459,4 +464,23 @@ export class Gaxios {
     }
     yield finale;
   }
+
+  /**
+   * A cache for the lazily-loaded proxy agent.
+   *
+   * Should use {@link Gaxios[#getProxyAgent]} to retrieve.
+   */
+  // using `import` to dynamically import the types here
+  static #proxyAgent?: typeof import('https-proxy-agent').HttpsProxyAgent;
+
+  /**
+   * Imports, caches, and returns a proxy agent - if not already imported
+   *
+   * @returns A proxy agent
+   */
+  static async #getProxyAgent() {
+    this.#proxyAgent ||= (await import('https-proxy-agent')).HttpsProxyAgent;
+
+    return this.#proxyAgent;
+  }
```

Comment:
This may have the convenient side-effect of preventing `esbuild` from being automatically pulled in!

See `out.js` via:
```sh
npx esbuild ./build/src/index.js --platform=node --bundle --outfile=out.js
```